addon/orm/associations/has-many.js

```diff
@@ -114,11 +114,11 @@ class HasMany extends Association {
           schema[association.target].where(query).update(foreignKey, null);
 
           // Save any children that are new
-          models.filter(model => model.isNew())
+          models.filter(model => model['isNew'] && model.isNew())
             .forEach(model => model.save());
 
           // Associate the new children to this model
-          schema[association.target].find(models.map(m => m.id)).update(foreignKey, this.id);
+          schema[association.target].find(models.map(m => m.id || m.attrs.id)).update(foreignKey, this.id);
```

Comment:
Some times that it was called I had to grab the attrs hash for the id. I'm really sorry but I didn't dig back to find the cause of it.